linkerd/service-profiles/src/lib.rs

```diff
@@ -25,6 +24,7 @@ pub mod http;
 mod proto;
 
 pub use self::client::Client;
+pub use linkerd_client_policy::{LogicalAddr, LookupAddr};
```

Comment:
I'd rather keep these things completely decoupled. Let's assume the logicaladdr and lookup types aren't going to be the same.

This PR should ~basically require no changes to anything in service-profiles (Though I think we'll need to cleanup some of the profile/discover stuff as a part of this work).